internal/executor/taskfilter/taskfilter.go

```diff
@@ -18,14 +18,61 @@ func MatchExactTask(desiredTaskName string) TaskFilter {
 		var filteredTasks []*api.Task
 
 		for _, task := range tasks {
-			if strings.EqualFold(desiredTaskName, task.Name) {
-				// Clear task's dependencies
-				task.RequiredGroups = task.RequiredGroups[:0]
+			// Ensure that this task's name matches with the name we're looking for
+			desiredTaskNameLower := strings.ToLower(desiredTaskName)
+			taskNameLower := strings.ToLower(task.Name)
 
-				filteredTasks = append(filteredTasks, task)
+			if !strings.HasPrefix(desiredTaskNameLower, taskNameLower) {
+				continue
 			}
+
+			// In case we're looking for a task with specific labels — extract them and ensure they all match
+			desiredLabels := extractLabels(strings.TrimPrefix(desiredTaskNameLower, taskNameLower))
+
+			var numMatchedLabels int
+
+			if task.Metadata != nil {
+				numMatchedLabels = matchLabels(desiredLabels, task.Metadata.UniqueLabels)
+			}
+
+			if numMatchedLabels != len(desiredLabels) {
```

Comment:
Maybe create a util method `containsAll` and have a single condition `containsAll(task.Metadata.UniqueLabels, desiredLabels)`?

Reply:
Wouldn't you still have two conditions, since we need to check for `task.Metadata != nil`?

Reply:
that's true but still IMO the code will be more readable than comparing how many labels matched. 🤔

Reply:
See b11cfb1e092c8f1b2407d24c01b581a13e04a2ea.